src/shared/components/post/post-listings.tsx

```diff
@@ -113,6 +139,49 @@ export class PostListings extends Component<PostListingsProps, any> {
     );
   }
 
+  handleHighlight(postIndex: number | undefined) {
+    postIndex != undefined && this.setState({ highlightedPost: postIndex });
+  }
+
+  toggleExpand() {
+    this.setState({ isExpanded: !this.state.isExpanded });
```

Comment:
State updates that depend on previous state should use functions to update.

```suggestion
    this.setState(prev => ({
        ...prev,
        isExpanded: !prev.isExpanded
    }));
```